parsers/src/main/scala/uk/gov/ons/addressIndex/crfscala/CrfFeature.scala

```diff
@@ -34,34 +34,29 @@ trait CrfFeature[T] {
     * @return
     */
   def toCrfJniInput(input: CrfToken, next: Option[CrfToken] = None, previous: Option[CrfToken] = None): CrfJniInput = {
-    StringBuilder
-      .newBuilder
-      .append(CrfScalaJni.lineStart)
-      .append(
-        createCrfJniInput(
-          prefix = name,
-          value = analyse(input)
-        )
-      )
-      .append(
-        next map { next =>
-          CrfScalaJni.delimiter +
+    val currentCrfJni: CrfJniInput = createCrfJniInput(
+      prefix = name,
+      value = analyse(input)
+    )
+    val nextCrfJni: CrfJniInput = {
+      next map { next =>
```

Comment:
Naming could improve if you are duplicating `next` it's hard to tell instantly which is the monad and which is the value.